autogluon/utils/tabular/ml/models/tabular_nn/tabular_nn_model.py

```diff
@@ -725,11 +727,13 @@ def save(self, file_prefix="", directory = None, return_name=False, verbose=None
             self.model.save_parameters(params_filepath)
         temp_model = self.model
         temp_sw = self.summary_writer
+        self._architecture_desc = self.model.architecture_desc
         self.model = None
         self.summary_writer = None
         save_pkl.save(path=modelobj_filepath, object=self, verbose=verbose)
         self.model = temp_model
         self.summary_writer = temp_sw
+        self._architecture_desc = None
```

Comment:
This is to save space?  
Are we sure this won't break repeated saving+reloading of the same model multiple times?

Reply:
This won't break anything, it is primarily to avoid an unnecessary variable. It does not save disk space. It saves a very minor amount of memory space.

This works by:

before save ->
self._architecture_desc = None

on save ->
set self._architecture_desc

on load ->
self._architecture_desc = None

Since self._architecture_desc val lives in .model, it can be retrieved from .model after load. self._architecture_desc is never used outside of save/load to init model